tests/test_l3_interface.py

```diff
@@ -44,93 +44,348 @@ def test_create_ip_dict_ipv4():
 
     # remove non existing address
-    new_state = nli.remove_ipv4_address(new_state, "203.0.113.7/24")
-    assert len(new_state["ipv4"]["addresses"]) == 2
-    assert new_state["ipv4"]["addresses"][0]["ip"] == ipv4_old.split("/")[0]
-    assert new_state["ipv4"]["addresses"][1]["ip"] == ipv4.split("/")[0]
+    ip4config = nli.remove_addresses(ip4config, ["203.0.113.7/24"])
+    assert len(ip4config["address"]) == 2
+    assert ip4config["address"][0]["ip"] == ipv4_old[0].split("/")[0]
+    assert ip4config["address"][1]["ip"] == ipv4[0].split("/")[0]
 
-    new_state = nli.remove_ipv4_address(new_state, ipv4_old)
-    assert len(new_state["ipv4"]["addresses"]) == 1
-    assert new_state["ipv4"]["addresses"][0]["ip"] == ipv4.split("/")[0]
+    ip4config = nli.remove_addresses(ip4config, ipv4_old)
+    assert len(ip4config["address"]) == 1
+    assert ip4config["address"][0]["ip"] == ipv4[0].split("/")[0]
 
 
-def test_set_and_remove_ipv4_address(base_state):
-    ipv4 = "198.51.100.31/24"
-    interface_state = nli.get_interface_state(base_state["interfaces"], "eth0")
-    new_state = nli.set_ipv4_addresses(interface_state, ipv4, False)
-    assert len(new_state["ipv4"]["addresses"]) == 1
-    assert new_state["ipv4"]["addresses"][0]["ip"] == ipv4.split("/")[0]
+def test_add_and_remove_ipv4_address(base_state):
+    ipv4 = ["198.51.100.31/24"]
+    ip4config = nli.add_addresses({}, ipv4)
+    assert len(ip4config["address"]) == 1
+    assert ip4config["address"][0]["ip"] == ipv4[0].split("/")[0]
+
+    ip4config = nli.remove_addresses(ip4config, ["203.0.113.7/24"])
+    assert len(ip4config["address"]) == 1
+    assert ip4config["address"][0]["ip"] == ipv4[0].split("/")[0]
+
+    ip4config = nli.remove_addresses(ip4config, ipv4)
+    assert len(ip4config["address"]) == 0
+
+    ip4config_b = nli.remove_addresses(ip4config, ipv4)
+    assert ip4config == ip4config_b
+
+
+def test_collect_single_settings():
+    aggregate = [
+        {"name": "eth0", "ipv4": "198.51.100.31/24", "ipv6": "2001:db8::1//32"}
+    ]
+    expected_config = {
+        "eth0": {"ipv4": ["198.51.100.31/24"], "ipv6": ["2001:db8::1//32"]}
+    }
+    interface_config = nli.collect_settings(aggregate)
+
+    assert interface_config == expected_config
+
+
+def test_single_addresses_present(base_state):
+    ansible_module = mock.Mock()
+    ansible_module.check_mode = False
+    ansible_module.params = dict(
+        name="eth0",
+        state="present",
+        ipv4="198.51.100.31/24",
+        ipv6="2001:db8::1/32",
+        purge=False,
+    )
+
+    with mock.patch(
+        "ansible_nmstate.netinfo.show", return_value=base_state
+    ) as mock_show:
+        nmstate_module = nli.AnsibleNMStateL3Interface(
+            ansible_module, "nmstate_interface_mock"
+        )
+        assert mock_show.called
+
+        with mock.patch("ansible_nmstate.netapplier.apply") as mock_apply:
+            nmstate_module.run()
+
+    assert mock_show.call_count == 2
```

Comment:
I think these needs to be indented to be inside the context of both mocks you use.

Reply:
it works this way. What appears to be the problem?

Reply:
Well, it looks outside the scope of its creation. It may be problematic depending on what you mock in. It just looked strange to check against it outside their relevant block scope.